src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -21,148 +87,1634 @@ const ActorInit En_Ossan_InitVars = {
+    this->cursorColorA = COL_CHAN_MIX(255, 0.0f, t);
+    this->cursorAnimTween = t;
+}
+
+void EnOssan_UpdateStickDirectionPromptAnim(EnOssan* this) {
+    f32 arrowAnimTween = this->arrowAnimTween;
+    f32 stickAnimTween = this->stickAnimTween;
+    s32 new_var2 = 255;
+    f32 new_var3;
+
+    if (this->arrowAnimState == 0) {
+        arrowAnimTween += 0.05f;
+        if (arrowAnimTween > 1.0f) {
+            arrowAnimTween = 1.0f;
+            this->arrowAnimState = 1;
+        }
+
+    } else {
+        arrowAnimTween -= 0.05f;
+        if (arrowAnimTween < 0.0f) {
+            arrowAnimTween = 0.0f;
+            this->arrowAnimState = 0;
+        }
+    }
+    this->arrowAnimTween = arrowAnimTween;
+
+    if (this->stickAnimState == 0) {
+        stickAnimTween += 0.1f;
+        if (stickAnimTween > 1.0f) {
+            stickAnimTween = 1.0f;
+            this->stickAnimState = 1;
+        }
+    } else {
+        stickAnimTween = 0.0f;
+        this->stickAnimState = 0;
+    }
+
+    new_var3 = 155.0f * arrowAnimTween;
+
+    this->stickAnimTween = stickAnimTween;
+
+    this->stickLeftPrompt.arrowColorR = COL_CHAN_MIX(255, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorG = COL_CHAN_MIX(new_var2, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorB = COL_CHAN_MIX(0, -100, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowColorR = (new_var2 - ((s32)new_var3)) & 0xFF;
+    this->stickRightPrompt.arrowColorG = (255 - ((s32)new_var3)) & 0xFF;
+    this->stickRightPrompt.arrowColorB = COL_CHAN_MIX(0, -100.0f, arrowAnimTween);
+    this->stickRightPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowTexX = 290.0f;
+    this->stickLeftPrompt.arrowTexX = 33.0f;
+
+    this->stickRightPrompt.stickTexX = 274.0f;
+    this->stickRightPrompt.stickTexX += 8.0f * stickAnimTween;
+    this->stickLeftPrompt.stickTexX = 49.0f;
+    this->stickLeftPrompt.stickTexX -= 8.0f * stickAnimTween;
+
+    this->stickLeftPrompt.arrowTexY = this->stickRightPrompt.arrowTexY = 91.0f;
+    this->stickLeftPrompt.stickTexY = this->stickRightPrompt.stickTexY = 95.0f;
+}
+
+void EnOssan_WaitForBlink(EnOssan* this) {
+    s16 decr = this->blinkTimer - 1;
+
+    if (decr != 0) {
+        this->blinkTimer = decr;
+    } else {
+        this->blinkFunc = EnOssan_Blink;
+    }
+}
+
+void EnOssan_Blink(EnOssan* this) {
+    s16 decr = this->blinkTimer - 1;
+    s16 eyeTextureIdxTemp;
+
+    if (decr != 0) {
+        this->blinkTimer = decr;
+        return;
+    }
+    eyeTextureIdxTemp = this->eyeTextureIdx + 1;
+    if (eyeTextureIdxTemp > 2) {
+        this->eyeTextureIdx = 0;
+        this->blinkTimer = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+        this->blinkFunc = EnOssan_WaitForBlink;
+    } else {
+        this->eyeTextureIdx = eyeTextureIdxTemp;
+        this->blinkTimer = 1;
+    }
+}
```

Comment:
I hate this. They had perfectly good blinking in OoT that this is a shadow of. I assume all the temps are required?

Reply:
Yep. It's actually a copy paste from OOT's EnOssan.